src/main/java/com/drzewek/wfrp_npc_generator/service/AuthorizationFilter.java

```diff
@@ -0,0 +1,61 @@
+package com.drzewek.wfrp_npc_generator.service;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+@Service
+public class AuthorizationFilter extends OncePerRequestFilter {
+
+    @Value("${jwt.secret}")
+    private String secret = "coś";
```

Comment:
to możesz zmienić. Jak pole bierze wartość z "value" to lepiej nie inicjalizować jej wartości. Aby była domyślna wartość wystarczy tak zapisać value: "${jwt.secret:default_secret}. Po ":" możesz wpisać wartość domyślną

Reply:
zmienione